test/ignore.spec.js

```diff
@@ -0,0 +1,20 @@
+import Alpine from 'alpinejs'
+import { wait } from '@testing-library/dom'
+
+global.MutationObserver = class {
+    observe() {}
+}
+
+test('x-ignore ignores initialization of any new components within the node', async () => {
+    document.body.innerHTML = `
+    <div x-ignore="true">
```

Comment:
Could we extend this/add new tests to include cases for

- `<div x-ignore>` (I'm assuming it will work the same as `x-ignore="true"` unless I'm missing something
- `<div x-ignore="false">` -> the component should actually initialise

Like I mentioned in an earlier comment, what happens when we've got `<div x-data="{}" x-ignore>`? I reckon in that case, the component shouldn't initialise, but whatever we decide there should be a test to validate the behaviour.

Reply:
- Yes, when i was implementing, i consider this and allow `x-ignore` without value as defaults to `x-ignore="true"`
- Yes `x-ignore="false"` will allow initialize the components within the applied node. 
- I have already included tests for each use cases. Let me know if all good.